dimod/include/dimod/quadratic_model.h

```diff
@@ -577,6 +577,10 @@ class BinaryQuadraticModel : public QuadraticModelBase<Bias, Index> {
     template <class B, class I, class T>
     void add_bqm(const BinaryQuadraticModel<B, I>& bqm,
                  const std::vector<T>& mapping) {
+        if (bqm.num_variables() == 0)
+            // nothing to do, other BQM is empty
```

Comment:
There still might be an offset
```
this->offset() += bqm.offset();
```

Reply:
Yep good catch.

Reply:
This should be fixed now. First convert the BQM if necessary, then add offset, then you're done if empty.